modules/mobile-agents/android/client/src/org/wso2/mdm/agent/services/Operation.java

```diff
@@ -0,0 +1,899 @@
+			                      .toString()
+			                      .equalsIgnoreCase(
+					                      resources.getString(R.string.intent_extra_decrypt))) {
+				doEncrypt = false;
+			} else if (!encryptData.isNull(resources.getString(R.string.intent_extra_function))) {
+				doEncrypt =
+						Boolean.parseBoolean(
+								encryptData.get(resources.getString(R.string.intent_extra_function))
+								           .toString());
+			}
+		} catch (JSONException e) {
+			Log.e(TAG, "Invalid JSON format." + e);
+		}
+
+		ComponentName admin = new ComponentName(context, AgentDeviceAdminReceiver.class);
+
+		if (doEncrypt &&
+		    devicePolicyManager.getStorageEncryptionStatus() != DevicePolicyManager.ENCRYPTION_STATUS_UNSUPPORTED &&
+		    (devicePolicyManager.getStorageEncryptionStatus() == DevicePolicyManager.ENCRYPTION_STATUS_INACTIVE)) {
+
+			devicePolicyManager.setStorageEncryption(admin, doEncrypt);
+			Intent intent = new Intent(DevicePolicyManager.ACTION_START_ENCRYPTION);
+			intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+			context.startActivity(intent);
+
+		} else if (!doEncrypt &&
+		           devicePolicyManager.getStorageEncryptionStatus() != DevicePolicyManager.ENCRYPTION_STATUS_UNSUPPORTED &&
+		           (devicePolicyManager.getStorageEncryptionStatus() == DevicePolicyManager.ENCRYPTION_STATUS_ACTIVE || 
+		           devicePolicyManager.getStorageEncryptionStatus() == DevicePolicyManager.ENCRYPTION_STATUS_ACTIVATING)) {
+			
+			devicePolicyManager.setStorageEncryption(admin, doEncrypt);
+		}
+
+		String status;
+		if (devicePolicyManager.getStorageEncryptionStatus() !=
+		    DevicePolicyManager.ENCRYPTION_STATUS_UNSUPPORTED) {
+			status = resources.getString(R.string.shared_pref_default_status);
+		} else {
+			status = resources.getString(R.string.shared_pref_false_status);
+		}
+
+		resultBuilder.build(code, status);
+
+	}
+
+	/**
+	 * Mute the device.
+	 * @param code        - Operation code.
+	 * @param requestMode - Request mode(Normal mode or policy bundle mode).
+	 */
+	private void muteDevice(String code) {
+		resultBuilder.build(code);
+
+		AudioManager audioManager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
+		audioManager.setStreamVolume(AudioManager.STREAM_RING, DEFAULT_VOLUME, DEFAULT_FLAG);
+	}
+
+	/**
+	 * Create web clip (Web app shortcut on device home screen).
+	 * @param code - Operation code.
+	 * @param data - Data required(Web app data).
+	 */
+	public void createWebClip(String code, String data) {
+		String appUrl = null;
+		String title = null;
+
+		try {
+			JSONObject webClipData = new JSONObject(code);
+			appUrl = (String) webClipData.get(resources.getString(R.string.intent_extra_identity));
+			title = (String) webClipData.get(resources.getString(R.string.intent_extra_title));
+		} catch (JSONException e) {
+			Log.e(TAG, "Invalid JSON format. " + e);
+		}
+
+		resultBuilder.build(data);
+
+		if(appUrl!=null && title!=null){
```

Comment:
Need to format this line to have proper white spaces between string literals and operators.